subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildScopeServices.java

```diff
@@ -289,10 +290,11 @@ public boolean isValid() {
             classLoaderHierarchyHasher);
     }
 
-    protected ScriptPluginFactory createScriptPluginFactory(BuildOperationExecutor buildOperationExecutor) {
+    protected ScriptPluginFactory createScriptPluginFactory(ScriptingLanguages scriptingLanguages, InstantiatorFactory instantiatorFactory, BuildOperationExecutor buildOperationExecutor) {
         DefaultScriptPluginFactory defaultScriptPluginFactory = defaultScriptPluginFactory();
+        ScriptPluginFactorySelector.ProviderInstantiator instantiator = new ScriptPluginFactorySelector.DefaultProviderInstantiator(instantiatorFactory.inject(this));
```

Comment:
I'd replace this explicit instantiation by a factory as suggested in db83629b9be313d3f5e4ee3d839a9f1431a5f224.